analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -36,6 +38,33 @@ public override bool ShouldExecute()
         return finder.HasPotentialNullableValueAccess;
     }
 
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var operationInstance = context.Operation.Instance;
+        if (operationInstance.Kind == OperationKindEx.PropertyReference
+            && operationInstance.ToPropertyReference() is { Instance: var instance } propertyReference
```

Comment:
The `var` decomposition is bit abusing the pattern matching when it doesn't have a null check. It's easier to track `property.Instance` 3 times.
`propertyReference` isn't wrong, this is just shorter and still good enough
```suggestion
            && operationInstance.ToPropertyReference() is { } property
```

Reply:
Ok for `property` instead of `propertyReference`.

I don't see why using pattern matching to deconstruct an object should be considered abuse.

In `switch` expressions, for instance, is quite natural and very readable:
```cs
string y = x switch
{
	(null, null) => "case 1",
	(var x1, var x2) when x1 == 4 && x2 == x1 + 1 => "case 2",
	_ => "case 3",
};
```

That is equivalent to 
```cs
string y;
if (x is (null, null)) y = "case 1";
else if (x is (var x1, var x2) && x1 == 4 && x2 == x1 + 1) y = "case 2";
else y = "case 3";
```

The most obvious way to match is to always match, which is what `var` does.

Reply:
In switch expression, there's no other way. 
In the case we had, I have to track one additional state `instance` that represents just indirection/rename for something that we already have like `property.Instance`. That makes the code harder to read and understand.

The `var` should be used only where there's no other way.

Reply:
So, if I understand it well, the indirection/rename is fine when it imposes some constraints on the capture (`case1` below), but it is not fine when it captures `instance` and only constraints the result of `ToPropertyReference` to be non-null, without constraining `Instance` to be non-null (`case2` below)?

```cs
case1: operationInstance.ToPropertyReference() is { Instance: { } instance }
case2: operationInstance.ToPropertyReference() is { Instance: var instance }
```

I fail to understand the logic behind such differentiation. Both pattern matching impose some constraints, with `case2` capturing a constrained property, whereas `case1` captures an unconstrained property (which is the goal of the var pattern feature in the first place).

The same indirection/rename also happens every time you create a local variable to store the result of a computation or casting to an object. The only difference is that the capture happens in an assignment statement.

```cs
var operationInstance = context.Operation.Instance;
// or
var switchExpression = (SwitchExpressionSyntaxWrapper)node;
var parent = (UnaryPatternSyntaxWrapper)node;
// etc.
```

Those cases are endemic in the `sonar-dotnet` code base.

I would argue that, in the `switch` expressions there is "no other way" just because pattern matching has been design to play together with deconstruction (an example of the design intents are explained by Neil Gafter in this post: https://github.com/dotnet/csharplang/issues/981#issuecomment-335952271), and that there is nothing wrong in `var` patterns.

And to conclude, I disagree with the statement that makes the code harder to read and understand.
It's easier for me to have a semantically well-defined variable for the rest of my code, rather than having property accesses like `property.Instance` scattered around. 
I think that is anyway a matter of personal taste, which should go beyond what should be presented as a required change in a pull request.

And on top, as mentioned in my previous comment, `.Instance` here is performing some computation, i.e. it is building a new wrapper every time you call it...

Reply:
There's a big difference between case 1 and 2. The `{ } instance` introduces a new state that is guaranteed to be not null and that changes the conditions under the condition is true. 
Contrary to `var instance` that doesn't have any functional effect and changes only the readability. I need to read a lot of syntax sugar to get something I already had and to learn that nothing happened. And I need to track one additional indirection. And that makes it harder to read.

For example `var` is fine here https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Rules/DoNotCallMethodsBase.cs#L42 where it's more practical than breaking it into 2 conditions and `var` statement.

Does it make sense?

Reply:
I'm not concerned about the new instance, as shimming is generally a fixed price we have to pay to do this. And there will be roughly very few additional instances created (approximately as few as issues).

Reply:
As discussed offline, I am changing both `ToPropertyReference` and `ToConversion` to store their results into dedicated variables (using `is var x`, which in this scenario is allowed) and calling `Instance` or `Operand` as many times as required.